### ISSUE TITLE
 mvn install FAILURE：Community DBMS Integration Tests and Community Bolt Integration Tests FAILURE

### ISSUE THREAD

**Issue by fubingting** [neo4j/neo4j#13375]
**Neo4j Version:** 5.14.0 and 5.15.0
**Operating System:** centos8 aarch64
**API:** maven@3.8.2 and openjdk@17.0.2

### Steps to reproduce
1.mvn install -fn  report error:
[INFO] Neo4j - Community DBMS Integration Tests ........... FAILURE [ 17.555 s]
[INFO] Neo4j - Community Graph Algorithms Integration Tests SUCCESS [ 16.236 s]
[INFO] Neo4j - Community Bolt Integration Tests ........... FAILURE [04:07 min]
2.mvn install -rf :dbms-it,The detailed error report is as follows：
```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.582 s -- in org.neo4j.commandline.dbms.AdminToolTest
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[ERROR] Tests run: 16, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.236 s <<< FAILURE! -- in org.neo4j.commandline.dbms.MigrateConfigCommandTest
[ERROR] org.neo4j.commandline.dbms.MigrateConfigCommandTest.suggestedJvmAdditionalsShouldMatchTemplate -- Time elapsed: 0.011 s <<< ERROR!
java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.toString()" because the return value of "java.nio.file.Path.getFileName()" is null
        at org.neo4j.commandline.dbms.MigrateConfigCommandTest.suggestedJvmAdditionalsShouldMatchTemplate(MigrateConfigCommandTest.java:392)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

[INFO] Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 15.73 s -- in org.neo4j.commandline.dbms.MemoryRecommendationsCommandTest
[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   MigrateConfigCommandTest.suggestedJvmAdditionalsShouldMatchTemplate:392 NullPointer Cannot invoke "java.nio.file.Path.toString()" because the return value of "java.nio.file.Path.getFileName()" is null
[INFO]
[ERROR] Tests run: 31, Failures: 0, Errors: 1, Skipped: 0

```
3.mvn install -rf :bolt-it，The detailed error report is as follows：
```
[ERROR] Tests run: 18, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 301.1 s <<< FAILURE! -- in org.neo4j.bolt.SchedulerBusyIT
[ERROR] org.neo4j.bolt.SchedulerBusyIT.shouldOperateNormallyWhenThreadsFreeUp(BoltWire, TransportConnection, TransportConnection, TransportConnection, TransportConnection)[5] -- Time elapsed: 61.36 s <<< ERROR!
org.awaitility.core.ConditionTimeoutException:
Assertion condition defined as a org.neo4j.test.assertion.Assert
expected: 2
 but was: 1 within 1 minutes.
        at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
        at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
        at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:985)
        at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:769)
        at org.neo4j.test.assertion.Assert.awaitUntilAsserted(Assert.java:56)
        at org.neo4j.test.assertion.Assert.awaitUntilAsserted(Assert.java:47)
        at org.neo4j.bolt.SchedulerBusyIT.awaitThreadPoolSaturation(SchedulerBusyIT.java:78)
        at org.neo4j.bolt.SchedulerBusyIT.shouldOperateNormallyWhenThreadsFreeUp(SchedulerBusyIT.java:162)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$15$1.accept(ReferencePipeline.java:541)
        at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:174)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
        Suppressed: org.neo4j.test.extension.timeout.VerboseTimeoutExceptionExtension$ThreadDump: Test 5.0 via UNIX-shouldOperateNormallyWhenThreadsFreeUp timed out. Full thread dump OpenJDK 64-Bit Server VM (17.0.2+8-86 mixed mode, sharing):

```
### Expected behavior
 mvn install -fn success

### Actual behavior
 mvn install -fn failure

**Comment by gjmwoods** [comment]
Hi @fubingting, many thanks for raising this issue.

Is your experience that these tests fail consistently? Due to the size of the test matrix we run there are inevitably some tests which fail periodically (ie they are flaky).

**Comment by fubingting** [comment]
> Hi @fubingting, many thanks for raising this issue.
> 
> Is your experience that these tests fail consistently? Due to the size of the test matrix we run there are inevitably some tests which fail periodically (ie they are flaky).

I tested many times on different machines and different systems, and found that Community DBMS Integration Tests will fail 100% and Community Bolt Integration Tests will fail occasionally.

**Comment by salted-yu** [comment]
Neo4j Version: 5.17
Operating System: Ubuntu 22.04
API: maven@3.9.6 and java 17.0.10 2024-01-16 LTS

I get the same error.

**Comment by gjmwoods** [comment]
@fubingting @salted-yu Apologies for the delay in responding.

We did find a bug in the community build of Neo4j for `MigrateConfigCommandTest` which has now been fixed and will be available in the next version of Neo4j (5.20).

As for `SchedulerBusyIT` we have recently made improvements to this tests and it should now be more robust.

Many thanks again for raising this issue. I will close this issue for now but please do reopen if you find the problem persists in the next release.
